x/marker/simulation/genesis.go

```diff
@@ -16,14 +18,19 @@ import (
 
 // Simulation parameter constants
 const (
-	MaxTotalSupply         = "max_total_supply"
+	MaxSupply              = "max_supply"
 	EnableGovernance       = "enable_governance"
 	UnrestrictedDenomRegex = "unresticted_denom_regex"
 )
 
-// GenMaxTotalSupply randomized Maximum amount of supply to allow for markers
-func GenMaxTotalSupply(r *rand.Rand) uint64 {
-	return r.Uint64()
+// GenMaxSupply randomized Maximum amount of supply to allow for markers
+func GenMaxSupply(r *rand.Rand) math.Int {
+	maxSupply := fmt.Sprintf("%d%d", r.Uint64(), r.Uint64())
+	res, ok := math.NewIntFromString(maxSupply)
+	if !ok {
+		panic("unable to generate max total supply")
+	}
```

Comment:
Also, can we cover this panic with a test?

Reply:
I ended up rewriting this to use the other function I wrote StringToBigInt. Looks a lot cleaner and that function has tests written for it to test this case.